services/user/Profiles/plugin/src/contact_table.rs

```diff
@@ -0,0 +1,98 @@
+pub const CONTACTS_KEY: &str = "contacts";
+
+use std::str::FromStr;
+
+use crate::bindings::clientdata::plugin::keyvalue as Keyvalue;
+use psibase::fracpack::{Pack, Unpack};
+use psibase::AccountNumber;
+
+use crate::bindings::host::common::types::Error;
+
+use crate::errors;
+use errors::ErrorType;
+
+#[derive(Pack, Unpack, Debug, Default)]
+pub struct ContactEntry {
+    pub account: AccountNumber,
+    pub nickname: Option<String>,
+    pub email: Option<String>,
+    pub phone: Option<String>,
+}
+
+impl ContactEntry {
+    pub fn new(
+        account: String,
+        nickname: Option<String>,
+        email: Option<String>,
+        phone: Option<String>,
+    ) -> Self {
+        Self {
+            account: AccountNumber::from_str(&account).expect("Invalid account number"),
+            nickname,
+            email,
+            phone,
+        }
+    }
+}
+
+pub struct ContactTable {
+    user: AccountNumber,
+}
+
+impl ContactTable {
+    pub fn new(user: AccountNumber) -> Self {
+        Self { user }
+    }
+
+    fn key(&self) -> String {
+        CONTACTS_KEY.to_string() + "." + &self.user.to_string()
```

Comment:
You actually don't even need `CONTACTS_KEY` here, your entire key can just be the current `user`.